pallets/settlement/src/lib.rs

```diff
@@ -1567,7 +1567,9 @@ impl<T: Config> Module<T> {
             Self::instruction_affirms_pending(id),
             Self::instruction_details(id).settlement_type,
         );
-        Self::prune_instruction(id);
+        if result.is_ok() {
+            Self::prune_instruction(id);
+        }
```

Comment:
We don't really need the `let result` variable, since this function doesn't return a value with `Ok`.

Something like this would also work:
```rust
Self::execute_settle_on_affirmation_instruction(
    id,
    Self::instruction_affirms_pending(id),
    Self::instruction_details(id).settlement_type,
)?;
// Prune the instruction if  it was executed successfully.
Self::prune_instruction(id);
Ok(())
```